### PR TITLE
update handleErr

### PR DESCRIPTION
**Pull request by gnehcein**

### CODE REVIEW

File: modules/tibc/core/types/genesis.go [hunk 1]
```diff
@@ -28,6 +30,9 @@ func (gs *GenesisState) Validate() error {
 	if err := gs.ClientGenesis.Validate(); err != nil {
 		return err
 	}
+	if err := gs.PacketGenesis.Validate(); err != nil {
+		return err
+	}
 
-	return gs.PacketGenesis.Validate()
+	return nil
```
Comment (chengwenxi) [inline modules/tibc/core/types/genesis.go:37]: What's the difference?

File: modules/tibc/core/04-packet/types/genesis.go [hunk 2]
```diff
@@ -43,12 +43,13 @@ func (ps PacketSequence) Validate() error {
 
 // NewGenesisState creates a GenesisState instance.
 func NewGenesisState(
-	acks, receipts, commitments []PacketState,
-	sendSeqs, recvSeqs, ackSeqs []PacketSequence, nextChannelSequence uint64,
+	acks, commitments, receipts []PacketState,
```
Comment (chengwenxi) [inline modules/tibc/core/04-packet/types/genesis.go:46]: Why switch the order?

File: modules/tibc/core/26-routing/keeper/keeper.go [hunk 3]
```diff
@@ -59,20 +58,25 @@ func (k Keeper) SetRoutingRules(ctx sdk.Context, rules []string) error {
...
+	}
+	return rules, nil
 }
 
 func (k Keeper) Authenticate(ctx sdk.Context, sourceChain, destinationChain, port string) bool {
-	rules, found := k.GetRoutingRules(ctx)
-	if !found {
+	rules, err := k.GetRoutingRules(ctx)
+	if err != nil {
+		panic(err)
```
Comment (chengwenxi) [inline modules/tibc/core/26-routing/keeper/keeper.go:77]: Return error for it, @dgsbl return error ack for this case.

File: modules/tibc/core/04-packet/types/genesis.go (hunk 2 above)
Comment (gnehcein) [reply modules/tibc/core/04-packet/types/genesis.go:46]: To be consistent with the order of the fields in GenesisState

File: modules/tibc/core/types/genesis.go (hunk 1 above)
Comment (gnehcein) [reply modules/tibc/core/types/genesis.go:37]: For the next update of routing's validate

File: modules/tibc/core/04-packet/types/genesis.go (hunk 2 above)
Comment (chengwenxi) [reply modules/tibc/core/04-packet/types/genesis.go:46]: If you do it, pls update `gengesis_test.go`.

File: modules/tibc/core/types/genesis.go (hunk 1 above)
Comment (chengwenxi) [reply modules/tibc/core/types/genesis.go:37]: Why not do it in the next update?

**Review by chengwenxi — APPROVED** [review]

File: modules/tibc/core/26-routing/keeper/keeper.go (hunk 3 above)
Comment (chengwenxi) [reply modules/tibc/core/26-routing/keeper/keeper.go:77]: return false

**Review by dgsbl — APPROVED** [review]

**Pull request merged**